File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    # 1) brown + yellow = w*h
    # 2) yellow = (w-2) * (h-2)
    
    total = brown + yellow # total = w*h
    for h in range(3, int(total ** 0.5) + 1): # total의 약수 h,w 쌍 구하기. yellow가 존재하려면 h가 무조건 3 이상
        if total % h == 0:
            w = total // h
            
            if (w-2) * (h-2) == yellow:
                return [w,h] # h를 최소값부터 보니까 w>=h 만족

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(int brown, int yellow) {
    int rc = brown + yellow;
    
    for (int r = 3; r <= sqrt(rc); r ++) {
        if (rc % r == 0) {
            int c = rc / r;
            
            if ((r-2) * (c-2) == yellow) return {c,r};
        }
    }
}